json-lint/schemas/inventory/placements/schema.json

```diff
@@ -5,8 +5,17 @@
   "title": "placements",
   "description": "placements",
   "additionalProperties": false,
-  "required": [
-    "adServer"
+  "oneOf": [
```

Comment:
unfortunately you can't require at least one... because entreprise can have an empty inventory.json
{
placements : {},
adservers: {}
}

Reply:
We should add also this case in the tests
https://github.com/Marfeel/DemoTenants/blob/829271cad13d5d0fe522024d6897d29197d6d30b/Tenants/vhosts/www.diabetesselfmanagement.com/index/resources/inventory.json

Reply:
This should be the case. If we have an empty inventory.json, the "placements" rule will check the matching properties with the "patternProperties" specified there **and then** proceed with the schema.json validation. Trivially, if the `"placements"` object is empty, the placement/schema.json will not be checked.

If what you say is true, this should have applied already as `"adServer"` in the placement/schema.json was already a required parameter (taking aside the fact the schema was never called from the main.json before).

Reply:
Validated offline, I'm right.

Reply:
can you validate this with a wrong test?

Reply:
@lucacamplimrf The wrong test should be there for the validation of the "oneOf" and it's already included in the `generic-placements.test.json`.

What I think we may want, to validate the empty inventory case, is a test in the **valid** ones. An empty inventory is valid and the required keys inside are not checked as the patternProperties in placements are not matched. I'll push the case for completeness.